File: app/Domains/User/DTOs/UpdateUserDTO.php
<?php

namespace App\Domains\User\DTOs;

class UpdateUserDTO
{
	public function __construct(
		private int $telegramID,
		private string $name,
		private string $photoUrl,
	){}

	public function getTelegramID(): int
	{
		return $this->telegramID;
	}

	public function getName(): string
	{
		return $this->name;
	}

	public function getPhotoUrl(): string
	{
		return $this->photoUrl;
	}
}

File: internal/dtos/update_user_dto.go
package dtos

type UpdateUserDTO struct {
	UserID   int64
	Name     string
	PhotoURL *string
}

func (dto UpdateUserDTO) GetUserID() int64 {
	return dto.UserID
}

func (dto UpdateUserDTO) GetName() string {
	return dto.Name
}

func (dto UpdateUserDTO) GetPhotoURL() *string {
	return dto.PhotoURL
}
